Kernel/Net/Socket.h

```diff
@@ -7,7 +7,7 @@
 #pragma once
 
 #include <AK/Error.h>
-#include <AK/Time.h>
+#include <AK/Forward.h>
```

Comment:
`Time` cannot be forward-declared for this header, it's directly used as members of this class.

Reply:
Oh it can, the compiler accepts it perfectly fine, and it reduces compilation time. It just leaves the struct incomplete, so that e.g. the size can't be determined.
Note that we already do this in many places, but okay, if you don't like it I'll remove it.